bundles/org.openhab.binding.tado/src/main/java/org/openhab/binding/tado/internal/handler/TadoZoneHandler.java

```diff
@@ -274,6 +319,50 @@ private void updateZoneState(boolean forceUpdate) {
         }
     }
 
+    /**
+     * Update the dynamic state descriptions for any channels which support an unknown sub- range of enumerator setting
+     * values, based on the list of capabilities reported by the respective zone.
+     *
+     * Note: currently this only applies to A/C devices that support fanLevel, horizontalSwing, or verticalSwing.
+     *
+     * @param zoneState the current zone Thing's state
+     */
+    private void updateDynamicStateDescriptions(ZoneState zoneState) {
+        GenericZoneSetting setting = zoneState.getSetting();
+        if (setting.getType() == TadoSystemType.AIR_CONDITIONING) {
```

Comment:
Consider inverting condition and returning fast to reduce nesting.

Reply:
Ok.